ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/RecoveredColumnSidecarSubscriber.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Consensys Software Inc., 2025
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.statetransition.datacolumns;
+
+import tech.pegasys.teku.spec.datastructures.blobs.DataColumnSidecar;
+import tech.pegasys.teku.statetransition.blobs.RemoteOrigin;
+
+public interface RecoveredColumnSidecarSubscriber {
+  void onRecoveredColumnSidecar(DataColumnSidecar dataColumnSidecar, RemoteOrigin origin);
```

Comment:
as we still have origin, it is confusing to have separate subscriber type. I'd reuse other DataColumnSidecar subscriber or default with RemoteOrigin.RECOVERED